finesse/gui/temp_control.py

```diff
@@ -346,10 +366,77 @@ def _create_controls(self) -> QGridLayout:
         layout.addWidget(self._set_sbox, 2, 1)
 
         self._update_pbtn = QPushButton("UPDATE")
+        self._update_pbtn.setCheckable(True)
+        self._update_pbtn.clicked.connect(self._on_update_clicked)
         layout.addWidget(self._update_pbtn, 2, 3)
 
         return layout
 
+    def _on_update_clicked(self) -> None:
+        isDown = self._update_pbtn.isChecked()
+        if isDown:
+            self._set_sbox.setEnabled(True)
+            self._end_polling()
+        elif not isDown:
+            self._set_new_set_point()
+            self._set_sbox.setEnabled(False)
+            self._begin_polling()
+
+    def _begin_polling(self) -> None:
+        """Initiate polling the TC4820 device."""
+        # SerialDevicePanel.set_controls_enabled will enable this, but
+        # want it to begin disabled
+        self._set_sbox.setEnabled(False)
```

Comment:
Hmm, that's slightly annoying behaviour on the part of `SerialDevicePanel`! I guess a workaround is fine for now though.

Reply:
Yeah, I thought about maybe having a special `set_controls_enabled` for this panel, but this seemed a simple enough solution.